src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Move.cs

```diff
@@ -92,6 +126,74 @@ private void DirectoryMove_DifferentWatchedDirectory(WatcherChangeTypes eventTyp
+            {
+                var dirs = Enumerable.Range(0, filesCount)
+                                .Select(i => new
+                                {
+                                    DirecoryInWatchedDir = Path.Combine(watchedTestDirectory.Path, $"dir{i}"),
+                                    DirecoryInUnwatchedDir = Path.Combine(unwatchedTestDirectory.Path, $"dir{i}")
+                                }).ToArray();
+
+                Array.ForEach(dirs, (dir) => Directory.CreateDirectory(dir.DirecoryInWatchedDir));
+
+                using (var watcher = new FileSystemWatcher(watchedTestDirectory.Path, "*"))
+                {
+                    Action action = () => Array.ForEach(dirs, dir => Directory.Move(dir.DirecoryInWatchedDir, dir.DirecoryInUnwatchedDir));
+
+                    // On macOS, for each file we receive two events as describe in comment below.
+                    int expectEvents = filesCount;
+                    if (skipOldEvents)
+                        expectEvents = expectEvents * 2;
+
+                    IEnumerable<FiredEvent> events = ExpectEvents(watcher, expectEvents, action);
+
+                    if (skipOldEvents)
+                        events = events.Where(x => x.EventType != WatcherChangeTypes.Created);
+
+                    var expectedEvents = dirs.Select(dir => new FiredEvent(WatcherChangeTypes.Deleted, dir.DirecoryInWatchedDir));
+
+                    // Remove Created events as there is racecondition when create dir and then observe parent folder. It receives Create event altought Watcher is not registered yet.
+                    Assert.Equal(expectedEvents, events.Where(x => x.EventType != WatcherChangeTypes.Created));
+                }
+            }
+        }
+
+        private void DirectoryMove_Multiple_FromUnwatchedToWatched(int filesCount)
+        {
+            Assert.True(filesCount > 0);
+
+            using (var watchedTestDirectory = new TempDirectory(GetTestFilePath()))
+            using (var unwatchedTestDirectory = new TempDirectory(GetTestFilePath()))
+            {
+
+                var dirs = Enumerable.Range(0, filesCount)
+                                .Select(i => new
+                                {
+                                    DirecoryInWatchedDir = Path.Combine(watchedTestDirectory.Path, $"dir{i}"),
+                                    DirecoryInUnwatchedDir = Path.Combine(unwatchedTestDirectory.Path, $"dir{i}")
+                                }).ToArray();
+
+                Array.ForEach(dirs, (dir) => Directory.CreateDirectory(dir.DirecoryInUnwatchedDir));
+
+                using (var watcher = new FileSystemWatcher(watchedTestDirectory.Path, "*"))
+                {
+                    Action action = () => Array.ForEach(dirs, dir => Directory.Move(dir.DirecoryInUnwatchedDir, dir.DirecoryInWatchedDir));
+
+                    var events = ExpectEvents(watcher, filesCount, action);
```

Comment:
We should avoid `var` here because the type returned by `ExpectEvents` is not obvious.

Reply:
This is what I really don't understand. When the type is simple then use the Type and when where is super obscure with anonymous classes `var` is fine?

Reply:
The idea is that code reviewers should be able to easily look at the variable and know its type immediately, without having to go find some method who's return value is being stored into it in order to figure it out. That means to use var, the type could be obvious from the RHS, namely a new expression or an explicit cast expression. Types where var is the only possible option (anonymous types) are obviously exempt, but they're also rare, and wouldn't show up in a line like this as they couldn't be the return type of this method.

Reply:
thx for explanation